Sources/System/FilePath/FilePathWindows.swift

```diff
@@ -461,3 +461,29 @@ extension SystemString {
     return lexer.current
   }
 }
+
+fileprivate let extendedLengthPathPrefix: [CInterop.PlatformChar] = [
+  UInt8(ascii: "\\"),
+  UInt8(ascii: "\\"),
+  UInt8(ascii: "?"),
+  UInt8(ascii: "\\")
+].map { .init($0) }
+
+extension UnsafePointer where Pointee == CInterop.PlatformChar {
+  /// On Windows, invokes `body` with a `\\?\` prefixed version of the pointee if it is a fully qualified DOS path.
+  ///
+  /// In all other cases, invokes the body with the original pointer.
+  ///
+  /// - precondition: The pointee should already have normalized path separators to `\`.
+  /// - seealso: https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation
+  internal func withExtendedLengthPrefixIfPossible<T>(_ body: (Self) -> T) -> T {
+    if ((self[0] >= UInt8(ascii: "A") && self[0] <= UInt8(ascii: "Z")) || (self[0] >= UInt8(ascii: "a") && self[0] <= UInt8(ascii: "z")))
+        && self[1] == UInt8(ascii: ":")
+        && self[2] == UInt8(ascii: "\\") {
+      return (extendedLengthPathPrefix + Array(UnsafeBufferPointer(start: self, count: system_platform_strlen(self) + 1))).withUnsafeBufferPointer { ptr in
```

Comment:
The way that I did this in Foundation was:

```swift
let dwLength = GetFullPathNameW(pwszPath, 0, nil, nil)
return try withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(dwLength)) {
  guard GetFullPathNameW(pwszPath, DWORD($0.count), $0.baseAddress, nil) > 0 else {
    throw CocoaError.errorWithFilePath(self, win32: GetLastError(), reading: true)
  }

  // 2. Perform the operation on the normalized path.
  return try body($0.baseAddress!)
}
```

This avoids us having to account for the various path normalization behaviours of Windows.

Reply:
Based on some offline discussion... I adopted your approach of using GetFullPathNameW, since (contrary to the documentation) it's capable of handling paths >260 chars regardless of the registry or process long paths enablement. However, we also need to call PathCchCanonicalizeEx to add the \\?\ prefix, since GetFullPathNameW will not do that, and the remaining Win32 path functions are expected to need it.

Reply:
In terms of any further high level business logic feedback, we should probably keep that on swiftlang/swift-foundation#1257 since I'm effectively copying that modified implementation over here to swift-system and swift-build as well.